api/envoy/api/v2/route/route.proto

```diff
@@ -1076,6 +1076,20 @@ message Tracing {
   // :ref:`HTTP Connection Manager <config_http_conn_man_runtime>`.
   // Default: 100%
   envoy.type.FractionalPercent overall_sampling = 3;
+
+  message CustomTag {
+    enum Type {
+      LITERAL = 0;
+      ENVIRONMENT = 1;
+      REQUEST_HEADER = 2;
+    }
+    Type type = 1 [(validate.rules).enum.defined_only = true];
+    string tag = 2;
+    string value = 3;
+    string default_value = 4;
```

Comment:
@objectiser i think i would be better to name this three field as `tag`, `value`, and `value_from`:
for type `LITERAL`, use `tag`:`value`
for others, use `tag`:`${value_from:value}`
what do you think?

Reply:
Not sure, as `default_value` identifies it as a fallback position.

Reply:
i'm also not quite sure, i just want introduce a reference semantic.

Reply:
provided different message configuration for each kind of source.